pkg/agent/datamodel/types.go

```diff
@@ -148,17 +277,17 @@ type CertificateProfile struct {
 
 // ServicePrincipalProfile contains the client and secret used by the cluster for Azure Resource CRUD
 type ServicePrincipalProfile struct {
-	ClientID          string                 `json:"clientId"`
-	Secret            string                 `json:"secret,omitempty" conform:"redact"`
-	ObjectID          string                 `json:"objectId,omitempty"`
-	KeyvaultSecretRef *api.KeyvaultSecretRef `json:"keyvaultSecretRef,omitempty"`
+	ClientID          string             `json:"clientId"`
+	Secret            string             `json:"secret,omitempty" conform:"redact"`
+	ObjectID          string             `json:"objectId,omitempty"`
+	KeyvaultSecretRef *KeyvaultSecretRef `json:"keyvaultSecretRef,omitempty"`
 }
 
 // JumpboxProfile describes properties of the jumpbox setup
 // in the AKS container cluster.
 type JumpboxProfile struct {
-	OSType    api.OSType `json:"osType"`
-	DNSPrefix string     `json:"dnsPrefix"`
+	OSType    OSType `json:"osType"`
+	DNSPrefix string `json:"dnsPrefix"`
 
```

Comment:
we don't need JumpbxoProfile. we can remove it

Reply:
Remove JumpboxProfile in this PR: https://github.com/Azure/AgentBaker/pull/240